packages/server/src/api/rest/index.ts

```diff
@@ -599,9 +606,18 @@ class RequestHandler {
 
         if (limit === Infinity) {
             const entities = await prisma[type].findMany(args);
+
+            let body = await this.serializeItems(type, entities, { include });
+            // Count items if totalCount is requested using only where filter, it not depends the others filter
+            const total = this.options.totalCount
```

Comment:
After looking at the original code in the "else" branch, I want to change my idea about using an option to opt-in the "count" behavior 😄 because it's counted anyway when pagination is enabled. Sorry for the change of mind. Maybe we remove the option for now?

And for this branch of fetching with `Infinity` limit, since `findMany` is already exhausting the list, I think we don't need to do a separate `count` query, and can just use `entities.length` for it, right?